bin/hledger-prices.hs

```diff
@@ -30,27 +32,59 @@ Print all prices from the journal.
 showPrice :: MarketPrice -> String
 showPrice mp = unwords ["P", show $ mpdate mp, T.unpack . quoteCommoditySymbolIfNeeded $ mpcommodity mp, showAmountWithZeroCommodity $ mpamount mp]
 
+divideAmount' :: Amount -> Quantity -> Amount
+divideAmount' a d = a' where
+    a' = (a `divideAmount` d) { astyle = style' }
+    style' = (astyle a) { asprecision = precision' }
+    extPrecision = (1+) . floor . logBase 10 $ realToFrac d
+    precision' = extPrecision + asprecision (astyle a)
+
```

Comment:
You are correct. In standard `divideAmount` precision inherited from first amount which effectively cuts off important information when price is inferred from ratio of amounts in different commodities.

I'd like to see standard amount routines to include `multiplyAmount` and/or maybe implement `Num` class. And probably should include log10 based arithmetic for estimating precision required for reverse transformation.

But I'm not sure about correctness of this implementation. It gives me result that banks in my country usually use to represent price. I.e. with inverted costs I get EUR price in UAH with mostly 4 digits after dot.